server-tasks/src/main/java/org/infinispan/tutorial/simple/server/tasks/InfinispanServerTasks.java

```diff
@@ -1,122 +1,59 @@
+import org.apache.commons.httpclient.HttpClient;
+import org.apache.commons.httpclient.methods.PostMethod;
+import org.apache.commons.httpclient.methods.StringRequestEntity;
 import org.infinispan.client.hotrod.RemoteCache;
 import org.infinispan.client.hotrod.RemoteCacheManager;
 import org.infinispan.client.hotrod.configuration.ConfigurationBuilder;
-import org.infinispan.commons.dataconversion.UTF8Encoder;
-import org.infinispan.commons.marshall.UTF8StringMarshaller;
-import org.infinispan.tasks.ServerTask;
-import org.infinispan.tasks.TaskContext;
+import org.infinispan.commons.api.CacheContainerAdmin;
 
-import java.util.Collections;
+import java.io.IOException;
 import java.util.HashMap;
 import java.util.Map;
-import java.util.stream.Collectors;
-import java.util.stream.IntStream;
+
+import static org.infinispan.commons.dataconversion.MediaType.APPLICATION_JAVASCRIPT_TYPE;
+import static org.infinispan.commons.util.Util.getResourceAsString;
 
 public class InfinispanServerTasks {
 
-   public static void main(String[] args) {
-      // Execute and data manipulation caches require different configuration
-      // This won't be necessary in the future: https://issues.jboss.org/browse/ISPN-8814
-      final RemoteCache<String, String> dataCache = getDataCache();
+   public static void main(String[] args) throws Exception {
+      // Upload the task using the REST API
+      uploadTask();
+
+      // Get a cache to execute the task
       final RemoteCache<String, String> execCache = getExecCache();
 
       // Create task parameters
       Map<String, String> parameters = new HashMap<>();
-      parameters.put("name", "developer");
+      parameters.put("greetee", "developer");
 
       // Execute hello task
       String greet = execCache.execute("hello-task", parameters);
       System.out.printf("Greeting = %s\n", greet);
 
-      // Store some values and compute the sum
-      int range = 10;
-      IntStream.range(0, range).boxed().forEach(
-         i -> dataCache.put(i + "-key", i + "-value")
-      );
-      int result = execCache.execute("sum-values-task", Collections.emptyMap());
-      System.out.printf("Sum of values = %d\n", result);
-
       // Stop the cache manager and release all resources
-      dataCache.getRemoteCacheManager().stop();
       execCache.getRemoteCacheManager().stop();
    }
 
-   public static RemoteCache<String,String> getDataCache() {
-      // Create a configuration for a locally-running server
-      ConfigurationBuilder builder = new ConfigurationBuilder();
-      builder
-         .addServer().host("127.0.0.1").port(11222)
-         .marshaller(UTF8StringMarshaller.class);
+   private static void uploadTask() throws IOException, InterruptedException {
+      String taskPostUrl = String.format("http://localhost:%d/rest/v2/tasks/hello-task", 11222);
+      String script = getResourceAsString("hello.js", InfinispanServerTasks.class.getClassLoader());
+      HttpClient client = new HttpClient();
```

Comment:
any reason for not using `org.infinispan.client.rest.RestClient`?

Reply:
because this is not required (the infinispan rest client)